README.md

```diff
@@ -23,6 +23,271 @@ pip install cdklabs.cdk-enterprise-iac
+
+In the example above, _all_ resources are created in the `appStack`, and an empty `extractedStack` is also created.
+
+We apply the `ResourceExtractor` Aspect, specifying the Cloudformation resource types the developer is unable to deploy due to insufficient IAM permissions.
+
+Now when we list stacks in the CDK project, we can see an added stack
+
+```zsh
+$ cdk ls
+MyAppStack
+ExtractedStack
+```
+
+Taking a look at these synthesized stacks, in the `ExtractedStack` we'll find:
+
+```yaml
+Resources:
+  TestLambdaServiceRoleC28C2D9C:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: 'sts:AssumeRole'
+            Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+        Version: 2012-10-17
+      # excluding remaining properties
+Outputs:
+  ExportAppStackTestLambdaServiceRoleC28C2D9C:
+    Value:
+      'Fn::GetAtt':
+        - TestLambdaServiceRoleC28C2D9C
+        - Arn
+    Export:
+      Name: 'AppStack:TestLambdaServiceRoleC28C2D9C'  # Exported name
+```
+
+And inside the synthesized `MyAppStack` template:
+
+```yaml
+Resources:
+  TestLambda2F70C45E:
+    Type: 'AWS::Lambda::Function'
+    Properties:
+      Role: !ImportValue 'AppStack:TestLambdaServiceRoleC28C2D9C'  # Using ImportValue instrinsic function to use pre-created IAM role
+      # excluding remaining properties
+```
+
+In this scenario, a developer is able to provide an external security team with sufficient IAM privileges to deploy the `ExtractedStack`.
+
+Once deployed, a developer can run `cdk deploy MyAppStack` without errors due to insufficient IAM privileges
+
+
+#### Value Sharing methods
+
+When resources are extracted from a stack, there must be a method to reference the resources that have been extracted.
+
+There are three methods (see `ResourceExtractorShareMethod` enum)
+
+- `CFN_OUTPUT`
+- `SSM_PARAMETER`
+- `API_LOOKUP`
+
+##### `CFN_OUTPUT`
+
+The default sharing method is `CFN_OUTPUT`, which uses Cloudformation Export/Import to Export values in the extracted stack (see [Outputs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/outputs-section-structure.html)), and use the [Fn::ImportValue](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-importvalue.html) intrinsic function to reference those values.
+
+This works fine, but some teams may prefer a looser coupling between the extracted stack deployed by an external team and the rest of the CDK infrastructure.
+
+##### `SSM_PARAMETER`
+
+In this method, the extracted stack generates Parameters in AWS Systems Manager Parameter Store, and modifies the CDK application to use [dynamic references](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html) to inject the [ssm parameter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/dynamic-references.html#dynamic-references-ssm) created in the extracted stack.
```

Comment:
This is actually not using Dynamic References - it is using ssm.StringParameter.fromLookup, which does an SDK API call to lookup the value at synthesis time and store it in the cdk.context.json file